index.php

```diff
@@ -99,7 +114,7 @@ function do_blocks( $content ) {
 	global $registered_blocks;
 
 	// Extract the blocks from the post content
-	$open_matcher = '/<!--\s*wp:([a-z](?:[a-z0-9\/]+)*)\s+((?:(?!-->).)*)-->.*<!--\s*\/wp:\g1\s+-->/';
+	$open_matcher = '/<!--\s*wp:([a-z](?:[a-z0-9\/]+)*)\s+((?:(?!-->).)*)--><!--\s*\/wp:\g1\s+-->/';
```

Comment:
I was thinking we should match only the empty blocks. see https://github.com/WordPress/gutenberg/pull/586#discussion_r114131814

But with the HTML fallback feature, we need to capture the `.* `.

--- 

Reintroducing the `.*` breaks the unit tests because this

```php
'<!-- wp:core/dummy value:b1 --><!-- /wp:core/dummy -->' .
'between' .
'<!-- wp:core/dummy value:b2 --><!-- /wp:core/dummy -->' .
```

match only one block, it takes the first and the last opening/closing tags only in consideration and all what's in between is catched by the `.*`. I'll take a deeper look and see if I can update the regex to change the priorities.